adagrams/game.py

```diff
@@ -1,13 +1,129 @@
 from random import randint
 
 def draw_letters():
-    pass
+    hand_length = 10
+    draw_pool = build_draw_pool()
+    hand = []
+    for i in range(hand_length):
+        random_num = randint( 0, len(draw_pool) - 1)
+        hand.append(draw_pool.pop(random_num))
+    return(hand)
 
 def uses_available_letters(word, letter_bank):
-    pass
+    uppercase_word = word.upper()
+    if not set(uppercase_word).issubset(letter_bank):
+        return False
+    copy_letter_bank = letter_bank.copy()
+    for letter in uppercase_word:
+        if letter in copy_letter_bank:
+            copy_letter_bank.remove(letter)
+        else:
+            return False
+    return True
 
 def score_word(word):
-    pass
+    if not word:
+        return 0
+    score_total = 0
+    if len(word) > 6:
+        score_total = 8
+
+    SCORE_LIST = {
+        'A': 1, 
+        'E': 1, 
+        'I': 1, 
+        'O': 1, 
+        'U': 1, 
+        'L': 1, 
+        'N': 1, 
+        'R': 1, 
+        'S': 1, 
+        'T': 1, 
+        'D': 2, 
+        'G': 2, 
+        'B': 3, 
+        'C': 3, 
+        'M': 3, 
+        'P': 3, 
+        'F': 4, 
+        'H': 4, 
+        'V': 4, 
+        'W': 4, 
+        'Y': 4, 
+        'K': 5, 
+        'J': 8, 
+        'X': 8, 
+        'Q': 10, 
+        'Z': 10
+        }
+    
+    uppercase_word = word.upper()
+    for letter in uppercase_word:
+        score_total += SCORE_LIST[letter]
+
+    return score_total
 
 def get_highest_word_score(word_list):
-    pass
\ No newline at end of file
+    winners_list, top_score = get_winners_list(word_list)
+    winner = winners_list[0]
+
+    if len(winners_list) > 1:
+        winner = do_tiebreak(winners_list)
+
+    return winner, top_score
+
+def do_tiebreak(winners_list):
+    shortest = winners_list[0]
+    for each in winners_list:
+        length = len(each)
+        if length == 10:
+            return each
+        if length < len(shortest):
+            shortest = each
+    return shortest
+
+def get_winners_list(word_list):
+    winners = []
+    top_score = 0
+    for each in word_list:
+        score = score_word(each)
+        if score > top_score:
+            top_score = score
+            winners = [each]
+        elif score == top_score:
+            winners.append(each)
+    return winners, top_score
+
+def build_draw_pool():
+    LETTER_POOL = {
+    'A': 9, 
+    'B': 2, 
+    'C': 2, 
+    'D': 4, 
+    'E': 12, 
+    'F': 2, 
+    'G': 3, 
+    'H': 2, 
+    'I': 9, 
+    'J': 1, 
+    'K': 1, 
+    'L': 4, 
+    'M': 2, 
+    'N': 6, 
+    'O': 8, 
+    'P': 2, 
+    'Q': 1, 
+    'R': 6, 
+    'S': 4, 
+    'T': 6, 
+    'U': 4, 
+    'V': 2, 
+    'W': 2, 
+    'X': 1, 
+    'Y': 2, 
+    'Z': 1
+    }
```

Comment:
To break up long lines of code, we should use multiple lines (see [here](https://peps.python.org/pep-0008/#code-lay-out) in the style guide about using multilines and indentation to create data structures).

```suggestion
LETTERS_POOL = {
    "A":9,"B":2,"C":2,"D":4,
    "E":12,"F":2,"G":3,"H":2,
    "I":9,"J":1,"K":1,"L":4,"M":2,
    "N":6,"O":8,"P":2,"Q":1,
    "R":6,"S":4,"T":6,"U":4,
    "V":2,"W":2,"X":1,
    "Y":2,"Z":1
}
```